javascripts/discourse/components/discourse-tag-banners.js

```diff
@@ -0,0 +1,86 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+
+export default class DiscourseTagBanners extends Component {
+  @service store;
+  @service router;
+  @service site;
+  @tracked tag = null;
+  @tracked keepDuringLoadingRoute = false;
+  @tracked isIntersection = false;
+
+  get isVisible() {
+    if (
+      this.currentRouteParams.tag_id &&
+      this.currentRouteParams.tag_id !== "none"
+    ) {
+      this.keepDuringLoadingRoute = true;
+      this.isLoading = false;
+      return true;
+    } else {
+      if (this.router.currentRoute.name.includes("loading")) {
+        return this.keepDuringLoadingRoute;
+      } else {
+        this.keepDuringLoadingRoute = false;
+        this.tag = null;
+        return false;
+      }
+    }
+  }
```

Comment:
```suggestion
  get isVisible() {
    if (
      this.currentRouteParams.tag_id &&
      this.currentRouteParams.tag_id !== "none"
    ) {
      this.keepDuringLoadingRoute = true;
      this.isLoading = false;
      return true;
    } else {
      if (this.router.currentRoute.name.includes("loading")) {
        return this.keepDuringLoadingRoute;
      } else {
        this.keepDuringLoadingRoute = false;
        this.tag = null;
        return false;
      }
    }
  }
```

1. `this.isLoading` seems unused?
2. this is a getter with side-effects 😢 getters shouldn't change other props